raiden/utils/cli.py

```diff
@@ -301,5 +310,78 @@ def expand_default(default, params):
         return HypenTemplate(default).substitute(params)
 
 
+def apply_config_file(
+    command_function: Union[click.Command, click.Group],
+    cli_params: Dict[str, Any],
+    ctx,
+    config_file_option_name='config_file',
+):
+    """ Applies all options set in the config file to `cli_params` """
+    paramname_to_param = {param.name: param for param in command_function.params}
+    path_params = {
+        param.name
+        for param in command_function.params
+        if isinstance(param.type, (click.Path, click.File))
+    }
+
+    config_file_path = Path(cli_params[config_file_option_name])
+    config_file_values = dict()
+    try:
+        with config_file_path.open() as config_file:
+            config_file_values = pytoml.load(config_file)
+    except OSError as ex:
+        # Silently ignore if 'file not found' and the config file path is the default
+        config_file_param = paramname_to_param[config_file_option_name]
+        config_file_default_path = Path(
+            config_file_param.type.expand_default(config_file_param.get_default(ctx), cli_params),
+        )
+        default_config_missing = (
+            ex.errno == errno.ENOENT and
+            config_file_path.resolve() == config_file_default_path.resolve()
+        )
+        if default_config_missing:
+            cli_params['config_file'] = None
+        else:
+            click.secho(f"Error opening config file: {ex}", fg='red')
+            sys.exit(2)
```

Comment:
We have usually used `1` for exiting the node, do you want to have better exit codes?

If yes, I think we should give these codes a name in the `constants.py`.

Reply:
+1 on Augusto's comment here. Please use `1` as an exit error code for now and in the future we can discuss appropriate codes and add them in `constants.py`

Reply:
Just FYI: We've been using 2 for config file related errors since that feature was introduced.